### ISSUE TITLE
Individual search on relationship field triggers error

### ISSUE THREAD

**Issue by Sicklou** [filamentphp/filament#3895]
### Package

filament/tables

### Package Version

v2.15.45

### Laravel Version

v9.28.0

### Livewire Version

v2.10.7

### PHP Version

8.1.0

### Problem description

When using individual search on a relationship field, defined by a dot notation, it triggers an error : 
`Filament\Tables\Concerns\{closure}(): Argument #1 ($search) must be of type string, array given`

It is thrown by the `CanSearchRecords::getTableColumnSearchQueries()`.

### Expected behavior

No errors, and the searching to work.

### Steps to reproduce

Adding a relationship field to a Table, then enabling the `isIndividual` parameter : 
```php
Columns\TextColumn::make('relation.field')
    ->label('Field')
    ->searchable(isIndividual: true);
```

Then start typing in the search field, and it should triggers the error.

### Reproduction repository

https://github.com/Sicklou/filament-individual-search-bug

**Comment by joaovdiasb** [comment]
dot notation is reserved from laravel, you should access as array using getStateUsing

`->getStateUsing(fn($record) => $record->relation['field'] ?? null)`

**Comment by danharrin** [comment]
@joaovdiasb dot notation is used for relationships and arrays in Filament. No need for `getStateUsing()`.

This issue is actually a bug with the individual column search feature that was introduced.

**Comment by joaovdiasb** [comment]
> @joaovdiasb dot notation is used for relationships and arrays in Filament. No need for `getStateUsing()`.
> 
> This issue is actually a bug with the individual column search feature that was introduced.

Good to know! I never managed to use it, I thought it could be some feature restriction

**Comment by danharrin** [comment]
Yes, and using dot notation will eager-load the relationship automatically

**Comment by danharrin** [comment]
Fixed by #3932
